localstack/testing/pytest/fixtures.py

```diff
@@ -923,27 +923,34 @@ class StackDeployError(Exception):
     def __init__(self, describe_res: dict, events: list[dict]):
         self.describe_result = describe_res
         self.events = events
-        super().__init__(
-            f"Describe output:\n{json.dumps(self.describe_result, cls=CustomEncoder)}\nEvents:\n{self.format_events(events)}"
-        )
+
+        encoded_describe_output = json.dumps(self.describe_result, cls=CustomEncoder)
+        if config.CFN_VERBOSE_ERRORS:
+            msg = f"Describe output:\n{encoded_describe_output}\nEvents:\n{self.format_events(events)}"
+        else:
+            msg = f"Describe output:\n{encoded_describe_output}\nFailing resources:\n{self.format_events(events)}"
+
+        super().__init__(msg)
 
     def format_events(self, events: list[dict]) -> str:
-        event_details = (
-            json.dumps(
-                {
-                    key: event.get(key)
-                    for key in [
-                        "LogicalResourceId",
-                        "ResourceType",
-                        "ResourceStatus",
-                        "ResourceStatusReason",
-                    ]
-                },
-                cls=CustomEncoder,
-            )
-            for event in events
-        )
-        return "\n".join(event_details)
+        formatted_events = []
+
+        chronological_events = sorted(events, key=lambda event: event["Timestamp"])
+        for event in chronological_events:
+            if not event["ResourceStatus"].endswith("FAILED") and not config.CFN_VERBOSE_ERRORS:
+                continue
+
+            formatted_events.append(self.format_event(event))
```

Comment:
I think reversing this statement might make a bit more readable by avoiding the negations